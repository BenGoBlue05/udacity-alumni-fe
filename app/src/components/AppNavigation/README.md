## AppNavigation Component
A component that is the navigator of the app, used in
containers/AppContainer container.


### Example

```js
<AppNavigation
  pathname={location.pathname}
  isMobile={isMobile}
  user={user}
  handleSearch={this.handleSearch}
  navIsActive={navIsActive}
  navLinks={navLinks}
  onToggleNav={this.handleToggleNav}
>
  {React.cloneElement(this.props.children, this.props)}
</AppNavigation>

```

### Props

| Prop             | Type     | Default     | Possible Values
| ---------------- | -------- | ----------- | -------------------
| **isMobile**     | Bool     |             |
| **children**     | Node     |             |
| **user**         | Object   |             |
| **handleSearch** | Func     |             | A callback function
| **navIsActive**  | Bool     |             |
| **navLinks**     | Array    |             |
| **onToggleNav**  | Func     |             | A callback function
| **pathname**     | String   |             |

### Other Information
