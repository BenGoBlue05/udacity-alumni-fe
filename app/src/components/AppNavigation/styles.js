import styled from 'styled-components';
import Header from 'grommet-udacity/components/Header';

export const StyledHeader = styled(Header)`
  border-bottom: 1px solid #dbe2e8;
  box-shadow: 0 2px 4px 0 rgba(46, 61, 73, 0.2);
  position: relative;
`;
