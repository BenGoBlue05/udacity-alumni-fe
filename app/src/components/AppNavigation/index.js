import React, { PropTypes } from 'react';
import { MobileNav, LogoImage, Navbar } from 'components';
import Title from 'grommet-udacity/components/Title';
import Anchor from 'grommet-udacity/components/Anchor';
import MenuIcon from 'grommet-udacity/components/icons/base/Menu';
import { StyledHeader } from './styles';

const AppNavigation = ({
  isMobile,
  children,
  user,
  handleSearch,
  navIsActive,
  navLinks,
  onToggleNav,
  pathname,
}) => (
  <div>
    {!isMobile &&
      <Navbar
        pathname={pathname}
        user={user}
        onSearch={handleSearch}
      />
    }
    {!isMobile && children}
    {isMobile &&
      <MobileNav
        pathname={pathname}
        user={user}
        navActive={navIsActive}
        onToggleNav={onToggleNav}
        navLinks={navLinks}
      >
        <StyledHeader
          direction="row"
          justify="between"
          large
          pad={{ horizontal: 'medium', between: 'small' }}
        >
          {!navIsActive &&
            <Title a11yTitle="Go Home">
              <Anchor href="/">
                <LogoImage />
              </Anchor>
            </Title>
          }
          <Title s
            tyle={navIsActive && { display: 'none' }}
            onClick={onToggleNav}
            a11yTitle="Open Menu Right"
          >
            <MenuIcon
              colorIndex="brand"
              size="medium"
              type="control"
            />
          </Title>
        </StyledHeader>
        {children}
      </MobileNav>
    }
  </div>
);

AppNavigation.propTypes = {
  isMobile: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  user: PropTypes.object.isRequired,
  handleSearch: PropTypes.func.isRequired,
  navIsActive: PropTypes.bool.isRequired,
  navLinks: PropTypes.array.isRequired,
  onToggleNav: PropTypes.func.isRequired,
  pathname: PropTypes.string.isRequired,
};

export default AppNavigation;
