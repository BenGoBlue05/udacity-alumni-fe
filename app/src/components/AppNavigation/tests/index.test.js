import AppNavigation from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<AppNavigation />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <AppNavigation />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
