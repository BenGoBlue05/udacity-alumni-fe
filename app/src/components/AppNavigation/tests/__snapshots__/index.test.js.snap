exports[`<AppNavigation /> should render with default props 1`] = `
<div>
  <Navbar />
</div>
`;
